src/test/java/com/conveyal/gtfs/loader/JDBCTableWriterTest.java

```diff
@@ -366,12 +382,48 @@ private static PatternDTO createSimpleRouteAndPattern(String routeId, String pat
     public void cannotCreateFrequencyForTimetablePattern() throws InvalidNamespaceException, IOException, SQLException {
-        PatternDTO simplePattern = createSimpleRouteAndPattern("900", "8", "The Loop");
+        PatternDTO simplePattern = createRouteAndSimplePattern("900", "8", "The Loop");
         TripDTO tripInput = constructFrequencyTrip(simplePattern.pattern_id, simplePattern.route_id, 6 * 60 * 60);
         JdbcTableWriter createTripWriter = createTestTableWriter(Table.TRIPS);
         createTripWriter.create(mapper.writeValueAsString(tripInput), true);
     }
 
+    /**
+     * When multiple patterns reference a single shape_id, the returned JSON from an update to any of these patterns
+     * (whether the shape points were updated or not) should have a new shape_id because of the "copy on update" logic
+     * that ensures the shared shape is not modified.
+     */
+    @Test
+    public void shouldChangeShapeIdOnPatternUpdate() throws IOException, SQLException, InvalidNamespaceException {
+        String patternId = "10";
+        ShapePointDTO[] shapes = new ShapePointDTO[]{
+            new ShapePointDTO(2, 0.0, sharedShapeId, firstStopLat, firstStopLon, 0),
+            new ShapePointDTO(2, 150.0, sharedShapeId, lastStopLat, lastStopLon, 1)
+        };
+        PatternStopDTO[] patternStops = new PatternStopDTO[]{
+            new PatternStopDTO(patternId, firstStopId, 0),
+            new PatternStopDTO(patternId, lastStopId, 1)
+        };
+        PatternDTO simplePattern = createRouteAndPattern("1001", patternId, "The Line", sharedShapeId, shapes, patternStops, 0);
+        assertThat(simplePattern.shape_id, equalTo(sharedShapeId));
+        // Create pattern with shared shape. Note: typically we would encounter shared shapes on imported feeds (e.g.,
+        // BART), but this should simulate the situation well enough.
+        String secondPatternId = "11";
+        patternStops[0].pattern_id = secondPatternId;
+        patternStops[1].pattern_id = secondPatternId;
+        PatternDTO patternWithSharedShape = createRouteAndPattern("1002", secondPatternId, "The Line 2", sharedShapeId, shapes, patternStops, 0);
+        // Verify that shape_id is shared.
+        assertThat(patternWithSharedShape.shape_id, equalTo(sharedShapeId));
+        // Update any field on one of the patterns.
+        JdbcTableWriter patternUpdater = createTestTableWriter(Table.PATTERNS);
+        patternWithSharedShape.name = "The shape_id should update";
+        String sharedPatternOutput = patternUpdater.update(patternWithSharedShape.id, mapper.writeValueAsString(patternWithSharedShape), true);
+        // The output should contain a new backend-generated shape_id.
+        PatternDTO updatedSharedPattern = mapper.readValue(sharedPatternOutput, PatternDTO.class);
+        LOG.info("Updated pattern output: {}", sharedPatternOutput);
+        assertThat(updatedSharedPattern.shape_id, not(equalTo(sharedShapeId)));
```

Comment:
We need to get into the habit of also checking that the database looks good in these tests.  Funky stuff could happen in the database that aren't reflected in the output from gtfs-lib.  Please add another few lines of code that make sure that the `shape_id` of the pattern with the secondPatternId doesn't equal the previous shared shape id.